doc/mgmt/SONiC_Ztp_DesignDoc.md

```diff
@@ -0,0 +1,282 @@
+
+**ztp run**
+
+Use this command to manually restart a new ZTP session. This command deletes the existing ```/etc/sonic/config_db.json``` file and stats ZTP service. It also erases the previous ZTP session data. The ZTP configuration is loaded on to the switch and ZTP discovery is performed. This command is useful to restart ZTP after it has failed or has been disabled by user.
+
+
+
+## 2.2 Functional Description
+
+After recieving the request from the client, via an RPC, the rest server will transfer the control to processAction method in the app module(inside traslib). This method will parse the target uri path and will branch to the corresponding function. These functions will call the python scripts in the host to perform ZTP related actions, like enable, disable ..etc. The response from the output of the script is propagated back to processAction method and is converted to json. The json message is sent back to the client.
+
+# 3 Design
+## 3.1 Overview
+
+
+## 3.2 DB Changes
+
+N/A
+### 3.2.1 CONFIG DB
+
+N/A
+
+### 3.2.2 APP DB
+
+N/A
+
+### 3.2.3 STATE DB
+
+TBD. 
+
+### 3.2.4 ASIC DB
+
+N/A
+
+### 3.2.5 COUNTER DB
+
+N/A
+
+## 3.3 Switch State Service Design
+### 3.3.1 Orchestration Agent
+
+N/A
+
+### 3.3.2 Other Process
+N/A
+
+## 3.4 SyncD
+N/A
+
+## 3.5 SAI
+N/A
+
+## 3.6 User Interface
+### 3.6.1 Data Models
+```
+ +--rw sonic-ztp
+       +--rw ZTP-STATUS
+       |  +--rw admin_mode?            boolean
+       |  +--rw service?               string
+       |  +--rw status?                string
+       |  +--rw source?                string
+       |  +--rw runtime?               string
+       |  +--rw timestamp?             yang:date-and-time
+       |  +--rw jsonversion?           string
+       |  +--rw activity_string?       string
+       |  +--rw CONFIG_SECTION_LIST* [sectionname]
+       |     +--rw sectionname     string
+       |     +--rw status?         string
+       |     +--rw runtime?        string
+       |     +--rw timestamp?      yang:date-and-time
+       |     +--rw exitcode?       uint32
+       |     +--rw ignoreresult?   boolean
+       +--rw ZTP-ENABLE
+          +--rw admin_mode?   boolean
+
+  rpcs:
+    +---x ztp-run
+    |  +--ro output
+    |     +--ro status?          int32
+    |     +--ro status-detail?   string
+    +---x ztp-disable
+       +--ro output
+          +--ro status?          int32
+          +--ro status-detail?   string
+
+```
+
+### 3.6.2 CLI
+#### 3.6.2.1 Configuration Commands
+
+**ztp enable**
+
+```
+sonic# ztp enable
+Success
+```
+
+**ztp disable**
+
+```
+sonic# ztp disable
+Success
+```
+
+**ztp run**
+
+```
+sonic# ztp run
```

Comment:
We may need a way to answer the confirmation question. Following would be the command.

ztp run [confirm]

Reply:
ztp run is implemented for right now